src/equations/ideal_glm_mhd_multiion_2d.jl

```diff
@@ -0,0 +1,859 @@
+            throw(DimensionMismatch("`gammas` and `charge_to_mass` have to be filled with at least one value"))
+
+        new(gammas, charge_to_mass, electron_pressure, c_h)
+    end
+end
+
+function IdealGlmMhdMultiIonEquations2D(; gammas, charge_to_mass,
+                                        electron_pressure = electron_pressure_zero,
+                                        initial_c_h = convert(eltype(gammas), NaN))
+    _gammas = promote(gammas...)
+    _charge_to_mass = promote(charge_to_mass...)
+    RealT = promote_type(eltype(_gammas), eltype(_charge_to_mass))
+    __gammas = SVector(map(RealT, _gammas))
+    __charge_to_mass = SVector(map(RealT, _charge_to_mass))
+
+    NVARS = length(_gammas) * 5 + 4
+    NCOMP = length(_gammas)
+
+    return IdealGlmMhdMultiIonEquations2D{NVARS, NCOMP, RealT,
+                                          typeof(electron_pressure)}(__gammas,
+                                                                     __charge_to_mass,
+                                                                     electron_pressure,
+                                                                     initial_c_h)
+end
+
+# Outer constructor for `@reset` works correctly
+function IdealGlmMhdMultiIonEquations2D(gammas, charge_to_mass, electron_pressure, c_h)
+    return IdealGlmMhdMultiIonEquations2D(gammas = gammas,
+                                          charge_to_mass = charge_to_mass,
+                                          electron_pressure = electron_pressure,
+                                          initial_c_h = c_h)
+end
+
+@inline function Base.real(::IdealGlmMhdMultiIonEquations2D{NVARS, NCOMP, RealT}) where {
+                                                                                         NVARS,
+                                                                                         NCOMP,
+                                                                                         RealT
+                                                                                         }
+    RealT
+end
+
+"""
+    initial_condition_weak_blast_wave(x, t, equations::IdealGlmMhdMultiIonEquations2D)
+
+A weak blast wave (adapted to multi-ion MHD) from
+- Hennemann, S., Rueda-Ramírez, A. M., Hindenlang, F. J., & Gassner, G. J. (2021). A provably entropy 
+  stable subcell shock capturing approach for high order split form DG for the compressible Euler equations. 
+  Journal of Computational Physics, 426, 109935. [arXiv: 2008.12044](https://arxiv.org/abs/2008.12044).
+  [DOI: 10.1016/j.jcp.2020.109935](https://doi.org/10.1016/j.jcp.2020.109935)
+"""
+function initial_condition_weak_blast_wave(x, t,
+                                           equations::IdealGlmMhdMultiIonEquations2D)
+    # Adapted MHD version of the weak blast wave from Hennemann & Gassner JCP paper 2020 (Sec. 6.3)
+    # Same discontinuity in the velocities but with magnetic fields
+    # Set up polar coordinates
+    RealT = eltype(x)
```

Comment:
```suggestion
    RealT = real(equations)
```
For consistency with line 126 below, I think this would make sense, no?

Reply:
So far, we have always used the input coordinate `x` in initial conditions for consistency since not all equations come with a real type

Reply:
Reverted in 8af20590c534fde774ebe5721e1ccac0248d6013.